manager/integration/tests/common.py

```diff
@@ -383,6 +389,106 @@ def finalizer():
+def statefulset(request):
+    statefulset_manifest = {
+        'apiVersion': 'apps/v1',
+        'kind': 'StatefulSet',
+        'metadata': {
+            'name': 'test-statefulset'
+        },
+        'spec': {
+            'selector': {
+                'matchLabels': {
+                    'app': 'test-statefulset'
+                }
+            },
+            'serviceName': 'test-statefulset',
+            'replicas': 2,
+            'template': {
+                'metadata': {
+                    'labels': {
+                        'app': 'test-statefulset'
+                    }
+                },
+                'spec': {
+                    'terminationGracePeriodSeconds': 10,
+                    'containers': [{
+                        'image': 'busybox',
+                        'imagePullPolicy': 'IfNotPresent',
+                        'name': 'sleep',
+                        'args': [
+                            '/bin/sh',
+                            '-c',
+                            'while true;do date;sleep 5; done'
+                        ],
+                        'volumeMounts': [{
+                            'name': 'pod-data',
+                            'mountPath': '/data'
+                        }]
+                    }]
+                }
+            },
+            'volumeClaimTemplates': [{
+                'metadata': {
+                    'name': 'pod-data'
+                },
+                'spec': {
+                    'accessModes': [
+                        'ReadWriteOnce'
+                    ],
+                    'storageClassName': DEFAULT_STORAGECLASS_NAME,
+                    'resources': {
+                        'requests': {
+                            'storage': size_to_string(
+                                           DEFAULT_VOLUME_SIZE * Gi)
+                        }
+                    }
+                }
+            }]
+        }
+    }
+
+    def finalizer():
+        api = get_apps_api_client()
+        api.delete_namespaced_stateful_set(
+            name=statefulset_manifest['metadata']['name'],
+            namespace='default', body=k8sclient.V1DeleteOptions())
+
+        for i in range(DEFAULT_POD_TIMEOUT):
+            ret = api.list_namespaced_stateful_set(namespace='default')
+            found = False
+            for item in ret.items:
+                if item.metadata.name == \
+                  statefulset_manifest['metadata']['name']:
+                    found = True
+                    break
+            if not found:
+                break
+            time.sleep(DEFAULT_POD_INTERVAL)
+        assert not found
+
+        # We need to generate the names for the PVCs on our own so we can
+        # delete them.
+        api = get_core_api_client()
+        client = get_longhorn_api_client()
+        for i in range(statefulset_manifest['spec']['replicas']):
+            pvc_name = (statefulset_manifest['spec']['volumeClaimTemplates'][0]
```

Comment:
Is there another way to associate PVC with the stateful set? It's a bit weird if we can only associate them using combined volume name. Is there any tag or fields in the PVC or statefulset we can associate them directly?

Reply:
At least seems you already have `get_statefulset_volume_names` below. You can move it up.